### PR TITLE
Add cosmosdb support

### LINKED ISSUE #23: Add cosmosdb support
**Is your feature request related to a problem? Please describe.**
Library doesn't support creating a CosmosDb context

**Describe the solution you'd like**
Adding support to create CosmosDb context and handling of its connection string

### PR DESCRIPTION
**Pull request by dgonzalezp**
[release] Add cosmos support 
Issue #23

### CODE REVIEW

File: source/Celerik.NetCore.Services/Core/ApiBuilder.cs [hunk 1]
```diff
@@ -168,6 +168,39 @@ public ApiBuilder(IServiceCollection services, IConfiguration config)
...
+
+            if (ServiceType == ApiServiceType.ServiceEF)
+            {
+                var connectionString = _config.GetConnectionString(
+                    ApiConfigKeys.CosmosDbConnectionStringName);
+
+                var cosmosConectionString = new CosmosDBConnectionString(connectionString);
+                var databaseName = _config.GetValue<string>(ApiConfigKeys.CosmosDbDatabaseName);
+
+                if (!string.IsNullOrEmpty(connectionString))
```
Comment (equisoide) [inline source/Celerik.NetCore.Services/Core/ApiBuilder.cs:194]: put cosmosConectionString  and databaseName  declarations inside IF

File: source/Celerik.NetCore.Services/Core/ApiBuilder.cs [hunk 2]
```diff
@@ -168,6 +168,39 @@ public ApiBuilder(IServiceCollection services, IConfiguration config)
...
+                throw new InvalidOperationException(
+                    ServiceResources.Get("ApiBuilder.MethodAlreadyCalled", nameof(AddCosmosDb))
+                );
+
+            if (ServiceType == ApiServiceType.ServiceEF)
+            {
+                var connectionString = _config.GetConnectionString(
+                    ApiConfigKeys.CosmosDbConnectionStringName);
+
+                var cosmosConectionString = new CosmosDBConnectionString(connectionString);
```
Comment (equisoide) [inline source/Celerik.NetCore.Services/Core/ApiBuilder.cs]: coNNection?

**Review by equisoide — COMMENTED** [review]
Create an issue and associate the issue with the Pull Request

File: source/Celerik.NetCore.Services/Core/ApiBuilder.cs (hunk 1 above)
Comment (dgonzalezp) [reply source/Celerik.NetCore.Services/Core/ApiBuilder.cs:194]: Yes! Its inside the if, line 186

File: source/Celerik.NetCore.Services/Core/ApiBuilder.cs (hunk 2 above)
Comment (dgonzalezp) [reply source/Celerik.NetCore.Services/Core/ApiBuilder.cs]: Fixed

**Comment by dgonzalezp** [comment]
[release]

File: source/Celerik.NetCore.Services/Core/ApiBuilder.cs (hunk 1 above)
Comment (equisoide) [reply source/Celerik.NetCore.Services/Core/ApiBuilder.cs:194]: Inside the second one

**Review by equisoide — DISMISSED** [review]

**Review by equisoide — APPROVED** [review]

**Pull request merged**